src/include/migraphx/op/unique.hpp

```diff
@@ -0,0 +1,340 @@
+        }
+
+        // min = 1 unique element; max = full dimension along the axis
+        // The three outputted Indices are just 1-D:
+        std::vector<shape::dynamic_dimension> d_out{{1, max_uniq_ct}};
+        shape sh_idx{shape::int64_type, {d_out}};
+
+        // Unique elements themselves aren't necessarily 1-D.. fix its output shape:
+        for(size_t idx = 1; t_axis != axis_none && t_axis + idx < dim_x; idx++)
+            d_out.push_back({lens_x[t_axis + idx], lens_x[t_axis + idx]});
+        shape sh_y = {sh_x.type(), {d_out}};
+
+        return {{sh_y, sh_idx, sh_idx, sh_idx}};
+    }
+
+    argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
+    {
+        size_t uniq_ct     = 0;
+        auto sh_x          = args.front().get_shape();
+        size_t dim_x       = sh_x.ndim();
+        auto lens_x        = sh_x.lens();
+        shape output_shape = dyn_out.computed_shape;
+        auto vec_ss        = output_shape.sub_shapes();
+        auto x_ct          = sh_x.elements();
+        shape sh_y         = {vec_ss[0].type(), {x_ct}};
+        shape sh_idx       = {vec_ss[1].type(), {x_ct}};
+        shape sh_x_idx     = {vec_ss[1].type(), {x_ct}};
+
+        argument res_y{sh_y};
+        argument res_y_idx{sh_idx};
+        argument res_x_rev_idx{sh_idx};
+        argument res_y_ct_idx{sh_idx};
+
+        std::vector<size_t> out_y_idx;
+        std::vector<size_t> out_x_rev_idx;
+        std::vector<size_t> out_y_ct;
+
+        // If axis is not none, for >1D tensors, we have to consider
+        // then, the uniqueness of chunks of sub-tensors: a subsequence of built-ins..
+        // For a built-in type, chunk_sz is of course = 1
+        size_t chunk_sz = 1;
+        int64_t t_axis  = axis;
+        if(axis != axis_none)
+        {
+            t_axis   = tune_axis(dim_x, t_axis, name());
+            chunk_sz = sh_x.strides()[t_axis];
+        }
+
+        visit_all(args.front(), res_y)([&](auto x, auto y_flat) {
+            using o_type = typename decltype(x)::value_type;
+            std::vector<o_type> x_in(x_ct);
+            auto itr = x_in.begin();
+            shape_for_each(sh_x, [&](auto, auto idx) { *itr++ = x[idx]; });
+
+            const auto& [y_indices, x_rev_indices, y_count] =
+                sorted ? sorted_uniq_indices(x_in, chunk_sz)
+                       : unsorted_uniq_indices(x_in, chunk_sz);
+
+            uniq_ct = y_indices.size();
+
+            // construct y from x[indices] in flattened form
+            // later we reshape y to the final shape..
+            for(size_t y_idx = 0, uniq_idx = 0; uniq_idx < uniq_ct; uniq_idx++)
+                for(size_t u_idx = y_indices[uniq_idx] * chunk_sz, ct = 0; ct < chunk_sz; ct++)
+                    y_flat[y_idx++] = x_in[u_idx++];
+
+            out_y_idx     = std::move(y_indices);
+            out_x_rev_idx = std::move(x_rev_indices);
+            out_y_ct      = std::move(y_count);
```

Comment:
Since these are const references then `std::move` will just do a copy. I think it might better to assign these values directly using `std::tie`:

```cpp
std::tie(out_y_idx, out_x_rev_idx, out_y_ct) =
    sorted ? sorted_uniq_indices(x_in, chunk_sz)
           : unsorted_uniq_indices(x_in, chunk_sz);
```

Reply:
That is a good catch, the const part & the ignored move(), thanks, and I agree -- I wish the compiler had warned. 
Will remove the references actually, as the intent is to automatically move stuff from calls to sorted_uniq_indices() and unsorted_unique_indices().

Reply:
> Will remove the references actually, as the intent is to automatically move stuff from calls

Why do you need to assign it to one variable and move it to another? Why not just assign into the variable directly with `std::tie`?

Reply:
std::tie will do an assignment operation to the existing variable. A move() is therefore preferred here.

Reply:
Upon testing, I think tie will in this case just move() as well, i.e. implicitly. So you are right. I can use tie() here and avoid the extra lines of code.. Thanks.